pkg/provider/certificate_authority.go

```diff
@@ -138,28 +177,29 @@ func (pc PeerCAConfig) Load() (*PeerCertificateAuthority, error) {
 			pc.CertPath, err)
 	}
 
-	i, err := idFromKey(c[len(c)-1].PublicKey)
+	i, err := idFromKey(c[0].PublicKey)
 	if err != nil {
 		return nil, err
 	}
 
 	return &PeerCertificateAuthority{
-		Cert: c[0],
-		ID:   i,
+		RestChain: c[1:],
```

Comment:
I wonder whether there is a way to avoid the positional arguments here. What if someone has the cert chain in the wrong order? But, should be fine for now.